File: Source Code/UnmanagedDemo/Source/GameObject.cpp
#include "stdafx.h"
#include "GameObject.h"


GameObject::GameObject(void)
{
	position[0] = 0.0f;
	position[1] = 0.0f;
	origin[0] = 0.0f;
	origin[1] = 0.0f;
	scale[0] = 1.0f;
	scale[1] = 1.0f;
	rotation = 0.0f;
	texture = NULL;
	srcRect = DemoRectangle();
	tint = DemoColour();
	destroyer = false;
	index = -1;
}


GameObject::~GameObject(void)
{
}

File: Source Code/.Net Demo/Source/GameObject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;
using Box2D.XNA;

namespace DotNetDemo
{
    public class GameObject
    {
        public Vector2 scale;
        public Vector2 position;
        public Vector2 origin;
        public float rotation;
        public Texture2D texture;
        public Rectangle srcRect;
        public Color tint;
        public bool destroyer;
        public int index;
        public Body physBody;

        public GameObject()
        {
            position = Vector2.Zero;
            origin = Vector2.Zero;
            scale = Vector2.One;
            rotation = 0.0f;
            texture = null;
            srcRect = new Rectangle();
            tint = new Color();
            destroyer = false;
            index = -1;
        }

        public float Width
        {
            get { return (scale.X * texture.Width); }
            set { scale.X = (value / texture.Width); }
        }

        public float Height
        {
            get { return (scale.Y * texture.Height); }
            set { scale.Y = (value / texture.Height); }
        }


    }
}
